packages/pwa-kit-create-app/scripts/trim-extensions.js

```diff
@@ -120,91 +130,104 @@ function processFile(filePath, plugins) {
         }
     }
 
+    const nodeContainsPluginReferences = (node) => {
+        return pluginRegex.test(source.slice(node.start, node.end))
+    }
+
     // Traverse AST and remove nodes guarded by plugin flags
     traverse(ast, {
         // Handle variable declarations like:
         // const HelloWorld = PLUGIN_NAME && loadable(...)
         VariableDeclaration(nodePath) {
-            nodePath.node.declarations.forEach((declaration) => {
-                if (declaration.init && declaration.init.type === 'LogicalExpression') {
-                    const shouldKeep = evaluateLogicalExpression(declaration.init)
-                    if (!shouldKeep) {
-                        // Extract import path if declaration is an import statement
-                        if (
-                            declaration.init.right &&
-                            declaration.init.right.type === 'CallExpression' &&
-                            declaration.init.right.callee &&
-                            declaration.init.right.callee.type === 'Identifier' &&
-                            declaration.init.right.callee.name === 'loadable'
-                        ) {
-                            // Extract the import path from the loadable call expression and add it to the list of candidates for removal later.
-                            const importArg = declaration.init.right.arguments[0]
+            if (nodeContainsPluginReferences(nodePath.node)) {
```

Comment:
We want to go in only when the node contains one or more plugin flags.

Reply:
See my comment above on potential optimization